### PR TITLE
Add short description of HardwareData API

### PR DESCRIPTION
**Pull request by fmuyassarov**
To describe the feature in https://github.com/metal3-io/baremetal-operator/pull/1099

### CODE REVIEW

**Comment by fmuyassarov** [comment]
@zaneb @andfasano @dtantsur

File: docs/api.md [hunk 1]
```diff
@@ -822,3 +822,23 @@ metadata:
...
+with the latest inspection data retrieved from Ironic. HardwareData holds the same name and namespace
+as its corresponding BareMetalHost resource. Currently, HardwareData doesn't have *Status* subresource
+but only the *Spec*, which we cover next.
+
+### HardwareData spec
+
+As you probably already noticed, the *Spec* of HardwareData is the same as [.Status.hardware](#hardware)
+of the HardwareData. However, this is only temporarily like that and eventually we will drop
+[.Status.hardware](#hardware) from BareMetalHost and only rely on HardwareData *Spec*. The reason
+for having duplication of inspection data at the  moment is to avoid breaking the existing users.
```
Comment (kashifest) [inline docs/api.md:844]: Bre
```suggestion
for having duplication of inspection data at the  moment is to avoid breaking the existing deployments.
```

**Review by kashifest — COMMENTED** [review]
/lgtm

File: docs/api.md [hunk 2]
```diff
@@ -822,3 +822,23 @@ metadata:
...
+BareMetalHost is deleted. Deprovisioning of the BareMetalHost should not trigger the deletion of
+HardwareData, but during next provisioning it can be re-created (with the same name and namespace)
+with the latest inspection data retrieved from Ironic. HardwareData holds the same name and namespace
+as its corresponding BareMetalHost resource. Currently, HardwareData doesn't have *Status* subresource
+but only the *Spec*, which we cover next.
+
+### HardwareData spec
+
+As you probably already noticed, the *Spec* of HardwareData is the same as [.Status.hardware](#hardware)
+of the HardwareData. However, this is only temporarily like that and eventually we will drop
```
Comment (furkatgofurov7) [inline docs/api.md:842]:
```suggestion
of the BareMetalHost. However, this is temporary and eventually we will drop
```

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**